api/lib/auth_functions.pl

```diff
@@ -41,6 +42,26 @@ sub read_json
     return decode_json($json);
 }
 
+# encode json data and write to file ($umask optional, default 022)
+# do: write_json ($filePath,$data,$umask)
+
+sub write_json {
+
+    my $filePath = shift || die 'No file path specified';
+    my $dirname = dirname($filePath);
+    if (! -d $dirname) {
+        die 'No directory to save the json file';
+    }
+    
+    my $data = shift || die 'No data to encode to json';
+    my $umask = shift || 022;
+
+    umask $umask;
+    open my $fh, ">", $filePath;
```

Comment:
I'd change to something like:
```
 open my $fh, ">", $filePath || return 0;
 ...
 return 1
```

Reply:
why do you want to return 1 on success and 0 on error :-?

Reply:
Yes, it's usually how functions are written: return true on success, false otherwise.
You could also return an object on success and undef on failure.

The semantic of values returned from functions is usually the opposite of shell exit code ;)